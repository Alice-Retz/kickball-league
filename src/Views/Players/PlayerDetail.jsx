import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { getPlayerById } from '../../Services/players';

export default function PlayerDetail({ match }) {
	const { playerId } = match.params;
	const [player, setPlayer] = useState(null);
	const [loading, setLoading] = useState(true);

	useEffect(() => {
		getPlayerById(playerId)
			.then((resp) => setPlayer(resp))
			.finally(() => setLoading(false));
	}, [playerId]);

	if (loading) return <h1>Loading player...</h1>;

	return (
		<>
			<Link to='/players' className='App-link'>
				Back to Players
			</Link>
			<p>
				<h1>{player.name}</h1>
			</p>
			<ul>
				<Link to={`/teams/${player.team_id}`} className='App-link'>
					<li>{player.teams.name}</li>
				</Link>
				<li>{player.position}</li>
			</ul>
		</>
	);
}
