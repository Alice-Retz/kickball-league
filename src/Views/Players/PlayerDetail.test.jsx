import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import PlayerDetail from './PlayerDetail';

it('should render the details of a designated player', async () => {
	const { container } = render(
		<MemoryRouter>
			<PlayerDetail match={{ params: { playerId: '3' } }} />
		</MemoryRouter>
	);

	screen.getByText('Loading player...');

	const playerName = await screen.findByText('Mike Comeaux', { exact: false });

	expect(playerName).toBeInTheDocument();
	expect(container).toMatchSnapshot();
});
