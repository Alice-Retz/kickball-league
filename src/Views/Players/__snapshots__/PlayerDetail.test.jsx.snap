// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the details of a designated player 1`] = `
<div>
  <a
    class="App-link"
    href="/players"
  >
    Back to Players
  </a>
  <h1>
    Mike Comeaux
  </h1>
  <ul>
    <a
      class="App-link"
      href="/teams/4"
    >
      <li>
        Red Stick Tigers
      </li>
    </a>
    <li>
      left winger
    </li>
  </ul>
</div>
`;
